### PR TITLE
feat(ui/rest): Update project specific vulnerability action

### PR DESCRIPTION
**Pull request by smrutis1**
Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*#1300*)
> * Did you add or update any new dependencies that are required for your change? - No

Issue: Closes: #1300 

* Need to test updating vulnerabilities with action field from both REST(Patch - `/projects/{projectId}/vulnerabilities` ) and UI(Project detail page -> Vulnerabilities -> Select Vulnerabilities to be updated -> Change).

> Have you implemented any additional tests? - No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/CommonVulnerabilityPortletUtils.java [hunk 1]
```diff
@@ -35,7 +37,8 @@
                                                    Function<T,String> valueGetter,
                                                    Function<T,String> dateGetter,
                                                    Function<T,String> authorGetter,
-                                                   Function<T,String> commentGetter){
+                                                   Function<T,String> commentGetter,
+                                                   Function<T,String> actionGetter){
```
Comment (ravi110336) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/CommonVulnerabilityPortletUtils.java:41]: Indentation after the closing braces.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 2]
```diff
@@ -1793,13 +1797,16 @@ private boolean addToVulnerabilityRatings(Map<String, Map<String, VulnerabilityR
 
             VulnerabilityCheckStatus vulnerabilityCheckStatus = vulnerabilityCheckStatusHistory.get(vulnerabilityCheckStatusHistory.size() - 1);
             VulnerabilityRatingForProject rating = vulnerabilityCheckStatus.getVulnerabilityRating();
+            String action = vulnerabilityCheckStatus.getAction();
```
Comment (ravi110336) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:1800]: Can vulnerabilityCheckStatus be null? if so please consider the null check.

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/vulnerabilityModal.jspf [hunk 3]
```diff
@@ -34,6 +35,16 @@
...
+						<div class="form-group d-none" id="rating">
+							<label for="rating-change-for-selected"><liferay-ui:message key="rating" />:</label>
+							<select id="rating-change-for-selected" class="form-control" name="<portlet:namespace/><%=VulnerabilityRatingForProject.class%>">
+								<sw360:DisplayEnumOptions type="<%=VulnerabilityRatingForProject.class%>" inQuotes="true" />
+							</select>
+						</div>
+						<div class="form-group d-none" id="action">
+							<label for="update-action-for-selected"><liferay-ui:message key="action" />:</label>
+							<input id="update-action-for-selected" data-name="action" name="<portlet:namespace/><%=VulnerabilityCheckStatus.class%>" type="text" class="form-control" minlength="2"" />
+						</div>
```
Comment (ravi110336) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/vulnerabilityModal.jspf:47]: Please remove the extra " in minlength.

**Review by ravi110336 — COMMENTED** [review]
Reviewed the code, please consider some minor comments.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java (hunk 2 above)
Comment (smrutis1) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java]: It can not be `null` considering prior checks

**Comment by smrutis1** [comment]
Thanks @ravi110336 . Have incorporated the review comments along with updated `action` to `projectAction`

**Comment by ravi110336** [comment]
Thanks @smrutis1 , for incorporating the changes tested the UI and works fine, except some minor observation.

1. when we click on the Vulnerabilities tab in the project detail page, it shows Processing.. in the screen and  once we change the action for a Vulnerabilities also the Processing.. text persist.

Screenshot from 2021-07-22 11-23-30


2. After we update the Vulnerabilities action for the project, the Vulnerabilities tab doesn't update the count when we click, but after the page reload it works fine.

Before the Page Reload:
Screenshot from 2021-07-22 11-34-42

_After the Page Reload:_ (**Processing**... shows here as well)
Screenshot from 2021-07-22 11-38-01

**Comment by smrutis1** [comment]
Thanks.
1. Unfortunately not able to reproduce.
2. Expected behavior.

**Comment by ravi110336** [comment]
Thanks, may be issue in my machine.
Also i have tested the REST API to update the project action and rating works fine. We can merge this.

Screenshot from 2021-07-22 17-08-17
Screenshot from 2021-07-22 17-08-24

**Pull request merged**